apis/tags/v1beta1/tagbinding_parentref.go

```diff
@@ -62,39 +69,51 @@ func (r *TagsTagBindingParentRef) SetExternal(ref string) {
 }
 
 func (r *TagsTagBindingParentRef) ValidateExternal(ref string) error {
-	gvk := r.GetGVK()
-	switch gvk {
-	case refs.ProjectGVK:
-		id := &refs.ProjectRef{
-			Name:      r.Name,
-			Namespace: r.Namespace,
-			External:  r.External,
-		}
-		if err := id.ValidateExternal(ref); err != nil {
-			return err
-		}
-		return nil
-	default:
-		return fmt.Errorf("format of %s external=%q was not known", gvk.Kind, ref)
+	_, id := r.resolveReference()
+	if id == nil {
+		return fmt.Errorf("unknown format for external reference %q", r.External)
 	}
+
+	if err := id.ValidateExternal(ref); err != nil {
+		return err
+	}
+	return nil
 }
 
 func (r *TagsTagBindingParentRef) Normalize(ctx context.Context, reader client.Reader, defaultNamespace string) error {
-	gvk := r.GetGVK()
-	switch gvk {
-	case refs.ProjectGVK:
-		id := &refs.ProjectRef{
-			Name:      r.Name,
-			Namespace: r.Namespace,
-			External:  r.External,
-		}
-		if err := id.Normalize(ctx, reader, defaultNamespace); err != nil {
-			return err
+	service, id := r.resolveReference()
+	if id == nil {
+		return fmt.Errorf("unknown format for external reference %q", r.External)
+	}
+
+	if err := id.Normalize(ctx, reader, defaultNamespace); err != nil {
+		return err
+	}
+
+	// Include the service qualification in case of ambiguities
+	r.External = "//" + service + "/" + id.GetExternal()
+
+	return nil
+}
+
+func (r *TagsTagBindingParentRef) resolveReference() (string, refs.Ref) {
+	if suffix, ok := strings.CutPrefix(r.External, "//storage.googleapis.com/"); ok {
```

Comment:
This function `resolveReference` has very similar logic to `parseParent` in the controller. This duplication should be avoided by creating a shared utility function for parsing parent resource strings.